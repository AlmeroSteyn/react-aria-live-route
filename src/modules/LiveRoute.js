import React from 'react';
import { Route } from 'react-router-dom';
import { LiveMessage } from 'react-aria-live';
import PropTypes from 'prop-types';

const LiveRoute = ({ liveMessage, 'aria-live': ariaLive, ...rest }) =>
  <div>
    <LiveMessage
      aria-live={ariaLive ? ariaLive : 'polite'}
      message={liveMessage}
    />
    <Route {...rest} />
  </div>;

//Use the same proptypes as react-router Router component with the added strings
LiveRoute.propTypes = {
  liveMessage: PropTypes.string.isRequired,
  'aria-live': PropTypes.string,
  computedMatch: PropTypes.object, // private, from <Switch>
  path: PropTypes.string,
  exact: PropTypes.bool,
  strict: PropTypes.bool,
  sensitive: PropTypes.bool,
  component: PropTypes.func,
  render: PropTypes.func,
  children: PropTypes.oneOfType([PropTypes.func, PropTypes.node]),
  location: PropTypes.object,
};

export default LiveRoute;
