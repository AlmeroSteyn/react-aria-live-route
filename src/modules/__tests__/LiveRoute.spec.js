import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import LiveRoute from '../LiveRoute';

describe('LiveRoute', () => {
  it('should render correctly via snapshot', () => {
    const mockComponent = () => <span>demo</span>;

    const wrapper = shallow(
      <LiveRoute
        path="/demo"
        component={mockComponent}
        liveMessage="Mock loaded"
        aria-live="assertive"
      />
    );

    const tree = shallowToJson(wrapper);
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly via snapshot for default aria live level', () => {
    const mockComponent = () => <span>demo</span>;

    const wrapper = shallow(
      <LiveRoute
        path="/demo"
        component={mockComponent}
        liveMessage="Mock loaded"
      />
    );

    const tree = shallowToJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
});
