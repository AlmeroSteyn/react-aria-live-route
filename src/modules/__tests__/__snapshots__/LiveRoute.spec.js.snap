// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LiveRoute should render correctly via snapshot 1`] = `
<div>
  <LiveMessage
    aria-live="assertive"
    message="Mock loaded"
  />
  <Route
    component={[Function]}
    path="/demo"
  />
</div>
`;

exports[`LiveRoute should render correctly via snapshot for default aria live level 1`] = `
<div>
  <LiveMessage
    aria-live="polite"
    message="Mock loaded"
  />
  <Route
    component={[Function]}
    path="/demo"
  />
</div>
`;
